libethereum/TransactionQueue.cpp

```diff
@@ -328,6 +312,14 @@ void TransactionQueue::drop(h256 const& _txHash)
         return;
 
     UpgradeGuard ul(l);
+    if (m_dropped.size() == c_maxDroppedTransactionCount)
+    {
+        LOG(m_loggerDetail) << "Dropped transaction list is at capacity ("
+                            << c_maxDroppedTransactionCount
+                            << "), removing dropped transaction from list (txhash: "
+                            << *m_dropped.begin() << ")";
+        m_dropped.erase(m_dropped.begin());
```

Comment:
This way it removes random transaction. As @chfast pointed out in the issue, it would be nicer to have some kind of LRU cache here, what do you think?

Reply:
> 
> 
> This way it removes random transaction. As @chfast pointed out in the issue, it would be nicer to have some kind of LRU cache here, what do you think?

Agreed, didn't go with that in the first place since it increases the storage requirements but I'll go ahead and add that in.

Reply:
I don't care so much any more. I think it is ok to merge it as it is and fill additional issue.

Reply:
I'm fine if we merge this as is, too